libsolidity/AST.cpp

```diff
@@ -505,5 +520,103 @@ void Literal::checkTypeRequirements()
 		BOOST_THROW_EXCEPTION(createTypeError("Invalid literal value."));
 }
 
+std::string ParamDescription::getName() const
+{
+	return m_description.first;
+}
+
+std::string ParamDescription::getType() const
+{
+	return m_description.second;
+}
+
+ASTPointer<ASTString> FunctionDescription::getDocumentation() const
+{
+	auto function = dynamic_cast<FunctionDefinition const*>(m_description.second);
+	if (function)
+		return function->getDocumentation();
+
+	return ASTPointer<ASTString>();
+}
+
+string FunctionDescription::getSignature() const
+{
+	return m_description.first->getCanonicalSignature(m_description.second->getName());
+}
+
+string FunctionDescription::getName() const
+{
+	return m_description.second->getName();
+}
+
+bool FunctionDescription::isConstant() const
+{
+	auto function = dynamic_cast<FunctionDefinition const*>(m_description.second);
+	if (function)
+		return function->isDeclaredConst();
+
+	return true;
+}
+
+vector<ParamDescription> const FunctionDescription::getParameters() const
```

Comment:
You probably want `vector<ParamDescription const>`. Otherwise assigning this to a non-const `vector<ParamDescription>` could copy the whole vector...